src/main/java/com/team/piickle/ballot/service/BallotService.java

```diff
@@ -1,8 +1,163 @@
+        }
+    }
+
+    private void validBallotTopic(String ballotTopicId) {
+        BallotTopic ballotTopic = ballotTopicRepository.findById(String.valueOf(ballotTopicId))
+                .orElseThrow(() -> new GeneralException("올바르지 않은 투표 주제 id 입니다."));
+    }
+
+    public List<BallotTopicResponseDto> getBallotTopicList(String userEmail) {
+        if (userEmail == null) {
+            List<BallotTopic> ballotTopicList = ballotTopicRepository.findAllByOrderByOrder(PageRequest.of(0, 4));
+            return ballotTopicList.stream()
+                    .map(ballotTopic -> new BallotTopicResponseDto(ballotTopic))
+                    .collect(Collectors.toList());
+        }
+        String userId = "";
+        if (!userRepository.findByEmail(userEmail).isEmpty()) {
+            userId = userRepository.findByEmail(userEmail).get().getId();
+        }
+        List<BallotResult> completedIds = ballotResultRepository.findByUserId(userId);
+        List<BallotTopicResponseDto> randomBallotTopicsExcludeCompleted = ballotTopicRepository.findByIdNinOrderByOrder(completedIds.stream()
+                .map(ballotTopic -> ballotTopic.getBallotTopicId())
+                .collect(Collectors.toList()), PageRequest.of(0, 4))
+                .stream()
+                .map(ballotTopic -> new BallotTopicResponseDto(ballotTopic))
+                .collect(Collectors.toList());
+        if (randomBallotTopicsExcludeCompleted.size() < 4) {
+            List<BallotTopicResponseDto> randomBallotTopicsCompleted = ballotTopicRepository.findByIdInOrderByOrder(completedIds.stream()
+                    .map(ballotTopic -> ballotTopic.getBallotTopicId())
+                    .collect(Collectors.toList()), PageRequest.of(0, 4 - randomBallotTopicsExcludeCompleted.size()))
+                    .stream()
+                    .map(ballotTopic -> new BallotTopicResponseDto(ballotTopic))
+                    .collect(Collectors.toList());;
+            return Stream.concat(randomBallotTopicsCompleted.stream(), randomBallotTopicsExcludeCompleted.stream())
+                    .collect(Collectors.toList());
+        }
+        return randomBallotTopicsExcludeCompleted;
+    }
+
+    private List<BallotTopic> getTopicListForGuest(String userId) {
+        if (userId == null) {
+            return ballotTopicRepository.findAllOrderByOrder(PageRequest.of(0, 4));
+        }
+        return new ArrayList<>();
+    }
+
+    public BallotStatusDto getBallotStatusAndUserSelect(String ballotTopicId, String userEmail) {
+        BallotTopic ballotTopic = ballotTopicRepository.findById(ballotTopicId)
+                .orElseThrow(() -> new GeneralException("올바르지 않은 투표 주제 id 입니다."));
+        List<BallotItem> ballotItems = ballotItemRepository.findByBallotTopicId(new ObjectId(ballotTopicId));
+        String userId = "";
+        if (!userRepository.findByEmail(userEmail).isEmpty()) {
+            userId = userRepository.findByEmail(userEmail).get().getId();
+        }
```

Comment:
```suggestion
        String userId = Optional.nullable(userEmail)
                .flatMap(email -> userRepository.findByEmail(email))
                .map(user -> user.getId()).orElse("");
```

이런식으로 바꿔 보는거 어땡
비슷한 로직 들도??